x-pack/plugins/beats_management/public/lib/tags.ts

```diff
@@ -80,19 +91,52 @@ export class TagsLib {
           .map((config: any) => config.id)
           .filter((id: string) => id !== 'other');
 
+        const picked = this.pickDeep(activeConfig, knownConfigIds);
+
         const convertedConfig = {
-          ...yaml.safeLoad(activeConfig.other),
-          ...pick(activeConfig, knownConfigIds),
+          ...yaml.safeLoad(activeConfig.other || '{}'),
+          ...picked,
         };
 
         return {
           type,
           description,
-          configs: [convertedConfig],
+          configs: [this.hydrateTokenizedObject(convertedConfig)],
         } as ConfigurationBlock;
       });
 
       return transformedTag;
     });
   }
+
+  private hydrateTokenizedObject(model: any) {
```

Comment:
I think these functions would be easier to reason about at a glance with short, dedicated comments explaining what they're doing from a high level. The name `hydrate` is a little obscure.